### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by elegaanz**
I went with a bash script because it was the most convenient thing to manipulate a git repository and the file system. I still need to implement CI to build the linter and run that script.

### CODE REVIEW

File: tests/ref/local+test+0.1.0.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[
+  {
+    "code": "manifest/package/license/type",
+    "file": "typst.toml",
+    "kind": "error",
+    "message": "The `license` field should be a string"
```
Comment (saecki) [inline tests/ref/local+test+0.1.0.json]: As discussed, I think it would make sense to add the `span` information to the JSON output.
Either as 1-based `line` and 1-based UTF-32 character numbers, or 0/1-basd line indices/numbers and 0-based UTF-8 character byte offsets.

**Comment by saecki** [comment]
I've added a few more test references, and found some edge cases with fatal errors.
From my side this is now good to go :)

**Comment by elegaanz** [comment]
Thanks, let's merge then :)
